# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libGLESv2/entry_points_utils.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libGLESv2/entry_points_utils.h
--- a/gfx/angle/checkout/src/libGLESv2/entry_points_utils.h	Fri Nov 08 16:41:42 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,86 +0,0 @@
-//
-// Copyright 2018 The ANGLE Project Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-//
-// entry_point_utils:
-//   These helpers are used in GLES entry point routines.
-
-#ifndef LIBGLESV2_ENTRY_POINT_UTILS_H_
-#define LIBGLESV2_ENTRY_POINT_UTILS_H_
-
-#include "angle_gl.h"
-#include "common/Optional.h"
-#include "common/PackedEnums.h"
-#include "common/angleutils.h"
-#include "common/mathutil.h"
-#include "libGLESv2/entry_points_enum_autogen.h"
-
-namespace gl
-{
-// A template struct for determining the default value to return for each entry point.
-template <EntryPoint EP, typename ReturnType>
-struct DefaultReturnValue;
-
-// Default return values for each basic return type.
-template <EntryPoint EP>
-struct DefaultReturnValue<EP, GLint>
-{
-    static constexpr GLint kValue = -1;
-};
-
-// This doubles as the GLenum return value.
-template <EntryPoint EP>
-struct DefaultReturnValue<EP, GLuint>
-{
-    static constexpr GLuint kValue = 0;
-};
-
-template <EntryPoint EP>
-struct DefaultReturnValue<EP, GLboolean>
-{
-    static constexpr GLboolean kValue = GL_FALSE;
-};
-
-// Catch-all rules for pointer types.
-template <EntryPoint EP, typename PointerType>
-struct DefaultReturnValue<EP, const PointerType *>
-{
-    static constexpr const PointerType *kValue = nullptr;
-};
-
-template <EntryPoint EP, typename PointerType>
-struct DefaultReturnValue<EP, PointerType *>
-{
-    static constexpr PointerType *kValue = nullptr;
-};
-
-// Overloaded to return invalid index
-template <>
-struct DefaultReturnValue<EntryPoint::GetUniformBlockIndex, GLuint>
-{
-    static constexpr GLuint kValue = GL_INVALID_INDEX;
-};
-
-// Specialized enum error value.
-template <>
-struct DefaultReturnValue<EntryPoint::ClientWaitSync, GLenum>
-{
-    static constexpr GLenum kValue = GL_WAIT_FAILED;
-};
-
-// glTestFenceNV should still return TRUE for an invalid fence.
-template <>
-struct DefaultReturnValue<EntryPoint::TestFenceNV, GLboolean>
-{
-    static constexpr GLboolean kValue = GL_TRUE;
-};
-
-template <EntryPoint EP, typename ReturnType>
-constexpr ANGLE_INLINE ReturnType GetDefaultReturnValue()
-{
-    return DefaultReturnValue<EP, ReturnType>::kValue;
-}
-}  // namespace gl
-
-#endif  // LIBGLESV2_ENTRY_POINT_UTILS_H_